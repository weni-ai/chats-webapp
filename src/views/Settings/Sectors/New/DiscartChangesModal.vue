<template>
  <UnnnicModalDialog
    :modelValue="showModal"
    size="sm"
    :title="$t('new_sector.discart.title')"
    :primaryButtonProps="{ text: $t('new_sector.discart.confirm') }"
    :secondaryButtonProps="{ text: $t('back') }"
    class="discart-changes-modal"
    data-testid="discart-changes-modal"
  >
    <p class="discart-changes-modal__text">
      {{ $t('new_sector.discart.hint') }}
    </p>
  </UnnnicModalDialog>
</template>

<script>
export default {
  name: 'DiscartChangesModal',
  props: {
    showModal: {
      type: Boolean,
      required: true,
    },
  },
};
</script>

<style lang="scss" scoped>
.discart-changes-modal {
  :deep(.unnnic-modal-dialog__container) {
    // use to centralize modal in connect iframe
    right: calc(55% - 200px);
    margin-top: -64px;
  }
  &__text {
    text-align: left;
    font-size: $unnnic-font-size-body-gt;
    font-family: $unnnic-font-family-secondary;
    line-height: 2 * $unnnic-font-size-body-md;
  }
}
</style>
