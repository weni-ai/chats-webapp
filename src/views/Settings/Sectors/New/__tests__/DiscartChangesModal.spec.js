import { mount } from '@vue/test-utils';

import DiscartChangesModal from '../DiscartChangesModal.vue';
import { beforeEach, describe } from 'vitest';

const createWrapper = () => {
  return mount(DiscartChangesModal, { props: { showModal: true } });
};

describe('DiscartChangesModal.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('should display the modal title', () => {
    const title = wrapper.find('[data-testid="discart-changes-modal"]');
    expect(title.text()).toContain(wrapper.vm.$t('new_sector.discart.title'));
  });
  it('should display the modal hint', () => {
    const title = wrapper.find('[data-testid="discart-changes-modal"]');
    expect(title.text()).toContain(wrapper.vm.$t('new_sector.discart.hint'));
  });
  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
