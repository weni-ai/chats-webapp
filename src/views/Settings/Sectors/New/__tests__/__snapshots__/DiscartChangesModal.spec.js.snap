// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DiscartChangesModal.vue > to match snapshot 1`] = `
"<section data-v-ab1febba="" data-v-7a1d255d="" class="unnnic-modal-dialog discart-changes-modal" data-testid="discart-changes-modal">
  <section data-v-ab1febba="" class="unnnic-modal-dialog__overlay" data-testid="modal-overlay"></section>
  <section data-v-ab1febba="" class="unnnic-modal-dialog__container unnnic-modal-dialog__container--sm" data-testid="modal-container">
    <!---->
    <section data-v-ab1febba="" class="unnnic-modal-dialog__container__body">
      <header data-v-ab1febba="" class="unnnic-modal-dialog__container__header">
        <section data-v-ab1febba="" class="unnnic-modal-dialog__container__title-container">
          <!---->
          <h1 data-v-ab1febba="" class="unnnic-modal-dialog__container__title-text" data-testid="title-text">Title</h1>
        </section>
        <!---->
      </header>
      <section data-v-ab1febba="" class="unnnic-modal-dialog__container__content">
        <p data-v-7a1d255d="" class="discart-changes-modal__text">Text</p>
      </section>
      <section data-v-ab1febba="" data-testid="actions-section" class="unnnic-modal-dialog__container__actions"><button data-v-d127031e="" data-v-ab1febba="" data-testid="secondary-button" class="unnnic-modal-dialog__container__actions__secondary-button unnnic-button unnnic-button--size-large unnnic-button--tertiary unnnic-modal-dialog__container__actions__secondary-button">
          <!---->
          <!---->
          <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label"> Back</span>
          <!---->
        </button><button data-v-d127031e="" data-v-ab1febba="" data-testid="primary-button" class="unnnic-modal-dialog__container__actions__primary-button unnnic-button unnnic-button--size-large unnnic-button--primary unnnic-modal-dialog__container__actions__primary-button">
          <!---->
          <!---->
          <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label"> Discart</span>
          <!---->
        </button></section>
    </section>
  </section>
</section>"
`;
